### PR TITLE
Add docs and quickrun script

### LINKED ISSUE #21: Document running epidisco w/ coclobas
Host docs currently at https://gist.github.com/smondet/a96d6050c865906d699d0e2637f7ced8 in this repo.

### LINKED ISSUE #23: Simplify running epidisco on GCP
Some ideas:
- [ ] Use `gcloud` CLI rather than web UI for launching the "GClouder" instance
- [ ] [Make our own GClouder image](https://cloud.google.com/compute/docs/images/export-image)/have a [startup script](https://cloud.google.com/compute/docs/startupscript) rather than using a base OS image and requiring the user to type commands in "Setup the Host" into the shell
- [ ] Automate the creation of an NFS filer (or, better: have tools pull/push data from/to Google cloud storage so that we don't require an NFS filer!)
- [ ] Automate the edits to `coclobas/master/tools/docker/configuration.env`
- [ ] Pull reference, dbSNP, and other data used by tools from [Google Genomics public data sets](https://cloud.google.com/genomics/v1/public-data)
- [ ] Create `run_pipeline.ml` for the user

### PR DESCRIPTION
**Pull request by ihodes**
This should make it quick and easy to get MVP Coclobas cluster up and
running, as well as submit an Epidisco workflow to it.

Fixes #23

Fixes #21

### CODE REVIEW

File: docs/disco.sh [hunk 1]
```diff
@@ -0,0 +1,200 @@
+trap 'exit' ERR
+
+export GCLOUD_ZONE="us-east1-c"
```
Comment (smondet) [inline docs/disco.sh]: This should be configurable, people do it (https://github.com/hammerlab/stratocumulus/pull/38)

File: docs/disco.sh [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+fi
+
+create () {
+    local boxname="$1"; shift
+    local path="`dirname \"$0\"`"
+    local path="`( cd \"$path\" && pwd )`"
+    local script="$0"
+    gcloud compute instances create $boxname \
+           --image-family ubuntu-1604-lts --image-project ubuntu-os-cloud \
+           --zone us-east1-c --scopes cloud-platform --boot-disk-size 40GB
```
Comment (smondet) [inline docs/disco.sh]: Zone also here :)

File: docs/disco.sh [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    gcloud compute firewall-rules create https-on-$boxname --allow tcp:443 \
+           --source-tags=$boxname --source-ranges 0.0.0.0/0
+    # ##*/ gets the basename of "script"
```
Comment (smondet) [inline docs/disco.sh]: Those sequences should have `set -e` or use `&&`.

File: docs/disco.sh [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    gcloud compute copy-files $path/${script##*/} $boxname:~
+}
+
+get-external-ip () {
+    local boxname=$(hostname)
+    local desc="gcloud compute instances describe $boxname"
+    local ip=$($desc | grep natIP | cut -d ':' -f 2 | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')
```
Comment (smondet) [inline docs/disco.sh:55]: Needs `set -o pipefail` (bash-only https://bclary.com/blog/2006/07/20/pipefail-testing-pipeline-exit-codes/)

File: docs/disco.sh [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    source /coclo/configuration.env
+    # This utility (https://github.com/cioc/gcloudnfs) is pre-installed on the
+    # Docker image.
+    gcloudnfs create --zone us-east1-c --project $project \
+              --network default --machine-type n1-standard-1 \
+              --server-name $NFS_SERVER_NAME \
+              --data-disk-name $NFS_SERVER_NAME-disk --data-disk-type pd-standard \
+              --data-disk-size $size
+    sudo mkdir -p /nfs-pool
+    sudo mount -t nfs $NFS_SERVER_NAME:/nfs-pool /nfs-pool
```
Comment (smondet) [inline docs/disco.sh:175]: Also should use `&&` or `set -e`.

File: docs/running-on-coclobas.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+`./please.sh` to get a list of all possible commands.
+
+### Set Up a Launch Box
+
+We need a server on GCloud to run our scheduler and Ketrew, our workflow
+engine. This box will serve as the interface between your workflow and GCloud's running
+of them.
+
+You'll need the GCloud command line utility, installed
+[from here](https://cloud.google.com/sdk/downloads#interactive).
```
Comment (smondet) [inline docs/running-on-coclobas.md:21]: “Don't forget to initialize it.“ ?

File: docs/disco.sh [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+## DOCKER_IMAGE is optional (default: hammerlab/biokepi-run):
+## The docker image to use for the Kubernetes jobs
+# export DOCKER_IMAGE=something/this:that
+
+## Usual Biokepi variables used to download Broad software:
+# export GATK_JAR_URL="http://example.com/GATK.jar"
+# export MUTECT_JAR_URL="http://example.com/Mutect.jar"
+EOF
+    gcloud config set compute/zone $GCLOUD_ZONE
```
Comment (smondet) [inline docs/disco.sh]: Not sure it's a good thing to globally set the zone for people's `gcloud` installations...

File: docs/running-on-coclobas.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+The first time the Coclobas server is started, it needs to create a the
+Kubernetes cluster. This may take about 5-10 minutes; you can check its status
+with `./disco.sh cluster-status`.
+
+If it says `Initializing` wait a bit and try again, if it says `Ready`, we can
+go on to the next step.
+
+You can visit the Google Container Engine WebUI watch your cluster being
+created in the "Instance Group" tab.
```
Comment (smondet) [inline docs/running-on-coclobas.md:90]: I think it's better to go to the Container Engine UI.
In the future, clusters may have more than one instance group.

File: docs/disco.sh [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+# meant to be sudo-executed
+setup-docker () {
+    apt-get install -y docker.io
+    docker pull hammerlab/coclobas
+    mkdir -p /tmp/coclo
+    chmod 777 /tmp/coclo
+    cp configuration.env /tmp/coclo
+    cp $0 /tmp/coclo
```
Comment (smondet) [inline docs/disco.sh]: `$0` does not seem to work in this case:

``` bash
 $ cat script 

f () { echo "Hey: /bin/bash." ; }
$*

 $ ./script f
Hey: /bin/bash.
```

File: docs/running-on-coclobas.md (hunk 8 above)
Comment (hammer) [reply docs/running-on-coclobas.md:90]: I think it's better to use the REST API to query the status; who wants to open a browser?

Comment (smondet) [reply docs/running-on-coclobas.md:90]: Yes `./disco.sh cluster-status` does that but, @ihodes wants something more spectacular :) 
Google's little "loading" GIF is so 90's!

File: docs/disco.sh (hunk 9 above)
Comment (smondet) [reply docs/disco.sh]: My bad, I messed up with `$` evaluation :)

**Comment by ihodes** [comment]
@seb back to you, thank you for the comments; I believe I addressed them all!

**Comment by ihodes** [comment]
Fixes #23

**Pull request merged**